### PR TITLE
[rom_ext] Correct the certificates written to flash

### PR DESCRIPTION
**Pull request by cfrantz**
1. Round the data lengths written to flash up to the next word.  This error was causing the certificates to fail their signature tests.
2. Eliminate the pathlen constraint from the fake CA certificate and allow certificate chains of arbitrary lenth.  This allows us to verify the certificate chain from CA->UDS->CDI0->CDI1.

### CODE REVIEW

File: sw/device/silicon_creator/manuf/keys/fake/fake_ca.conf [hunk 1]
```diff
@@ -15,5 +15,5 @@ CN=Google Engineering ICA
 [v3_ca]
 subjectKeyIdentifier = hash
 authorityKeyIdentifier = keyid:always,issuer:always
-basicConstraints = critical,CA:true,pathlen:0
+basicConstraints = critical,CA:true,pathlen:3
```
Comment (timothytrippel) [inline sw/device/silicon_creator/manuf/keys/fake/fake_ca.conf:18]: pathlen of 4 to enable CDI_1 to endorse another cert in test flows?

**Review by timothytrippel — APPROVED** [review]
nice find!

File: sw/device/silicon_creator/manuf/keys/fake/fake_ca.conf (hunk 1 above)
Comment (cfrantz) [reply sw/device/silicon_creator/manuf/keys/fake/fake_ca.conf:18]: https://stackoverflow.com/questions/6616470/certificates-basic-constraints-path-length

Perhaps we should just omit the pathlen constraint from the fake CA.

Comment (cfrantz) [reply sw/device/silicon_creator/manuf/keys/fake/fake_ca.conf]: I eliminated the `pathlen` constraint.  We should still seek guidance on this when generating the real endorsement certificate to be used in production.

**Comment by cfrantz** [comment]
The CI test failure is not related to this PR.

**Pull request merged**